fsmonitor-settings.c

```diff
@@ -59,9 +59,12 @@ static int check_deprecated_builtin_config(struct repository *r)
 	 * the check resulted the config being set by this (deprecated) setting.
 	 */
 	if(!repo_config_get_bool(r, "core.useBuiltinFSMonitor", &core_use_builtin_fsmonitor)) {
-		advise_if_enabled(ADVICE_USE_CORE_FSMONITOR_CONFIG,
-				  _("core.useBuiltinFSMonitor will be deprecated "
-				    "soon; use core.fsmonitor instead"));
+		if (!git_env_bool("GIT_SUPPRESS_USEBUILTINFSMONITOR_ADVICE", 0)) {
+			advise_if_enabled(ADVICE_USE_CORE_FSMONITOR_CONFIG,
+					  _("core.useBuiltinFSMonitor will be deprecated "
+					    "soon; use core.fsmonitor instead"));
+			setenv("GIT_SUPPRESS_USEBUILTINFSMONITOR_ADVICE", "1", 1);
+		}
```

Comment:
Why use the environment when a `static int` suffices?

```suggestion
		static int gave_advice = 0;
		if (!gave_advice) {
			advise_if_enabled(ADVICE_USE_CORE_FSMONITOR_CONFIG,
					  _("core.useBuiltinFSMonitor will be deprecated "
					    "soon; use core.fsmonitor instead"));
			gave_advice = 1;
		}
```

Reply:
I was thinking about child processes, e.g. clean/smudge or diff filters, or hooks. There, a `static` won't suffice.